server/api/admin/challs/put.js

```diff
@@ -21,17 +23,36 @@ export default {
       type: 'object',
       properties: {
         data: {
-          type: 'object'
+          type: 'object',
+          properties: {
+            files: {
+              type: 'array'
+            }
+          }
         }
       }
     }
   },
   handler: async ({ req }) => {
+    const uploadProvider = getUploadProvider()
     const chall = req.body.data
 
     // Ensure id is consistent
     chall.id = req.params.id
 
+    const files = await Promise.all(
+      chall.files.map(async ({ name, data }) => {
+        const url = await uploadProvider.upload(toBuffer(data))
```

Comment:
We should catch any errors from `toBuffer` as it will throw if an invalid data URI is passed.